plugins/apps/threatbus_zmq_app/README.md

```diff
@@ -0,0 +1,167 @@
+...
+```
+
+Initially, apps that want to connect with this plugin only need to know the
+`manage` endpoint. The plugin and the app negotiate all other internals for
+pub-sub message exchange at runtime. See the protocol definition below for
+details.
+
+## Management Protocol
+
+Subscriptions and unsubscriptions are referred to as `management`. All
+management messages are JSON formatted and exchanged via the `manage` ZMQ
+endpoint, that the plugin exposes.
+
+The manage endpoint uses the
+[ZeroMQ Request/Reply](https://learning-0mq-with-pyzmq.readthedocs.io/en/latest/pyzmq/patterns/client_server.html)
+pattern for message exchange. That means, all messages get an immediate response
+from the server. With each JSON reply, the zmq-app Threat Bus plugin sends a
+`status` field that indicates the success of the requested operation.
+
+### Subscribe at the Plugin
+
+To subscribe to Threat Bus via the zmq-app plugin, an app needs to send a JSON
+struct as follows to the `manage` endpoint of the plugin:
+
+```
+{
+  "action": "subscribe",
+  "topic": <TOPIC>,       # either 'threatbus/sighting' or 'threatbus/intel'
+  "snapshot": <SNAPSHOT>  # number of days for a snapshot, 0 for no snapshot
+}
+```
+In response, the app will either receive a `success` or `unsuccess` response.
+
+- Unsuccess response:
+  ```
+  {
+    "status": "unsuccess"
+  }
+  ```
+- Success response:
+  ```
+  {
+    "topic": <P2P_TOPIC>,
+    "pub_endpoint": 13371,
+    "sub_endpoint": 13372,
+    "status": "success",
+  }
+  ```
+
+  The `pub_endpoint` and `sub_endpoint` are the port numbers that are configured
+  in the Threat Bus plugin config. The app can access these endpoints following
+  the [ZeroMQ Pub/Sub](https://learning-0mq-with-pyzmq.readthedocs.io/en/latest/pyzmq/patterns/pubsub.html)
+  pattern. The plugin will publish messages on the `pub_endpoint` and accept
+  messages on the `sub_endpoint`.
+
+  The `topic` field of the response contains a unique topic for the client. That
+  topic _must_ be used to receive messages. The unique topic is a 32 characters
+  wide random string and guarantees that other subscribers won't accidentally
+  see traffic that should only be visible to the new client.
+  
+  For more details see below at `Pub/Sub via ZeroMQ`.
+
+### Unsubscribe from the Plugin
+
+To unsubscribe a connected app, it has to send a JSON struct to the `manage`
+endpoint of the plugin, as follows:
+
+```
+{
+  "action": "unsubscribe",
+  "topic": <P2P_TOPIC>       # the 32-characters random topic that the got during subscription handshake
+}
+```
+
+In response, the app will either receive a `success` or `unsuccess` response.
+
+- Unsuccess response:
+  ```
+  {
+    "status": "unsuccess"
+  }
+  ```
+- Success response:
+  ```
+  {
+    "status": "success"
+  }
+  ```
+
+### Pub/Sub via ZeroMQ
+
+Once an app has subscribed to Threat Bus via using the `manage` endpoint of the
+zmq-app plugin, it got a unique, random `p2p_topic` (see above). Threat Bus
```

Comment:
```suggestion
zmq-app plugin, it has a unique, random `p2p_topic` (see above). Threat Bus
```

For technical documentation, let's be precise as possible and avoid jargon if we can.